storage/fanout.go

```diff
@@ -231,22 +232,38 @@ func NewMergeQuerier(primaryQuerier Querier, queriers []Querier) Querier {
 func (q *mergeQuerier) Select(params *SelectParams, matchers ...*labels.Matcher) (SeriesSet, Warnings, error) {
 	seriesSets := make([]SeriesSet, 0, len(q.queriers))
 	var warnings Warnings
-	for _, querier := range q.queriers {
-		set, wrn, err := querier.Select(params, matchers...)
-		q.setQuerierMap[set] = querier
-		if wrn != nil {
-			warnings = append(warnings, wrn...)
-		}
-		if err != nil {
-			q.failedQueriers[querier] = struct{}{}
-			// If the error source isn't the primary querier, return the error as a warning and continue.
-			if querier != q.primaryQuerier {
-				warnings = append(warnings, err)
-				continue
-			} else {
-				return nil, nil, err
+	wg := sync.WaitGroup{}
+	wg.Add(len(q.queriers))
+	queryResult := make(map[int]SeriesSet)
+	var priErr error
+	isPriError := false
+	for idx, qr := range q.queriers {
+		go func(index int, querier Querier) {
```

Comment:
If there's no remote storage configured do we end up creating goroutines?

Reply:
Thank you for your review. I have modified the code according to your suggestion .